skfp/preprocessing/filters/ghose.py

```diff
@@ -0,0 +1,95 @@
+from typing import Optional
+
+from rdkit.Chem import Mol
+from rdkit.Chem.Crippen import MolLogP, MolMR
+from rdkit.Chem.Descriptors import MolWt
+from rdkit.Chem.rdMolDescriptors import CalcNumAtoms
+
+from skfp.bases.base_filter import BaseFilter
+
+
+class GhoseFilter(BaseFilter):
+    """
+    Ghose rule.
+
+    Used to searching for drug-like  molecules [1]_.
+
+    Molecule must fulfill conditions:
+
+        - 160 <= molecular weight <= 400
+        - -0.4 <= logP <= 5.6
+        - 20 <= Natoms <=70
+        - 40 <= refractivity <= 130
+
+    Parameters
+    ----------
+    allow_one_violation : bool, default=True
+        Whether to allow violating one of the rules for a molecule. This makes the
+        filter less restrictive, and is the part of the original definition of this
+        filter.
+
+    n_jobs : int, default=None
+        The number of jobs to run in parallel. :meth:`transform_x_y` and
+        :meth:`transform` are parallelized over the input molecules. ``None`` means 1
+        unless in a :obj:`joblib.parallel_backend` context. ``-1`` means using all
+        processors. See Scikit-learn documentation on ``n_jobs`` for more details.
+
+    batch_size : int, default=None
+        Number of inputs processed in each batch. ``None`` divides input data into
+        equal-sized parts, as many as ``n_jobs``.
+
+    verbose : int, default=0
+        Controls the verbosity when generating conformers.
+
+    References
+    -----------
+    .. [1] `Ghose, A. K., Viswanadhan, V. N., & Wendoloski, J. J. (1999).
+        "A Knowledge-Based Approach in Designing Combinatorial or Medicinal Chemistry Libraries for Drug Discovery. 1.
+        A Qualitative and Quantitative Characterization of Known Drug Databases."
+        Journal of Combinatorial Chemistry, 1(1), 55–68.
+        <https://doi.org/10.1021/cc9800071>`_
+
+    Examples
+    ----------
+    >>> from skfp.preprocessing import GhoseFilter
+    >>> smiles = ["CC(=O)C1=C(O)C(=O)N(CCc2c[nH]c3ccccc23)C1c1ccc(C)cc1", "CC(=O)c1c(C)n(CC2CCCO2)c2ccc(O)cc12",\
+    "CC(=O)c1c(C(C)=O)c(C)n(CCCCn2c(C)c(C(C)=O)c(C(C)=O)c2C)c1C"]
+    >>> filt = GhoseFilter()
+    >>> filt
+    GhoseFilter()
+    >>> filtered_mols = filt.transform(smiles)
+    >>> filtered_mols
+    ['CC(=O)C1=C(O)C(=O)N(CCc2c[nH]c3ccccc23)C1c1ccc(C)cc1', 'CC(=O)c1c(C)n(CC2CCCO2)c2ccc(O)cc12']
+    """
+
+    def __init__(
+        self,
+        allow_one_violation: bool = False,
+        return_indicators: bool = False,
+        n_jobs: Optional[int] = None,
+        batch_size: Optional[int] = None,
+        verbose: int = 0,
+    ):
+
```

Comment:
Remove empty line

Reply:
Also check for all further filters. We don't want empty lines at the beginning of methods